packit/api.py

```diff
@@ -922,6 +946,23 @@ def sync_release(
                 )
                 self.up.set_specfile(self.dg.specfile)
 
+            spec_ver = self.up.get_specfile_version()
```

Comment:
I hope we don't break anything by this move of code..;)

Reply:
yes me too, this was my yesterday fear and I can not avoid it.

Reply:
and why is this move needed?

Reply:
we need to checkout the specfile from the dist-git-branch before doing any action to check against the version inside the specfile

Reply:
@majamassarini I don't see an issue in running this after `post_upstream_clone` that should be run before any Packit action. People sometimes generate a specfile in this step. (But we don't check the upstream specfile so that point should be fine.)

But with your change, there is no difference between `post_upstream_clone` and `pre_push`.

Reply:
Should this work only for pull from upstream? Now it is not like that. 
If it should work also for propose downstream and a specfile can be generated after `post_upstream_clone` then I have to move the `check_version_distance` after this action.

What I have changed moving the code `switch_branch` code is the ` self.dg.check_last_commit()` call. This call was always made against `rawhide` and now is made against the `dist-git-branch`. If we want to run the actions (both `post_upstream_clone` and `pre_sync`) even though the release should not be made then I can move the code back more or less as it was previously.

But my thought was that now we export distgit to be used in an action during the release synchronization and we were exporting the wrong distgit (always rawhide instead of the target branch); also for this reason I thought we should switch the distgit branch before any action is run.

Reply:
> Should this work only for pull from upstream? Now it is not like that.

Do you mean this feature? I would target both.

> But my thought was that now we export distgit to be used in an action during the release synchronization and we were exporting the wrong distgit (always rawhide instead of the target branch); also for this reason I thought we should switch the distgit branch before any action is run.

OK. This makes sense. In both cases, we need to make this clear in docs.

Reply:
Since `self.dg.check_last_commit()`  no one uses, we now have two actions with the same effect what might be a bit confusing.

Reply:
I am not against the change. We can also move the version check bellow the checkout and take a look at the actions later and make it also clear in docs.